File: backend/app/models/vehiculo.py
from sqlalchemy import Column, Integer, String, ForeignKey, DateTime
from sqlalchemy.orm import relationship
from app.models.base import Base
from datetime import datetime

class Vehiculo(Base):
    __tablename__ = "vehiculos"

    id = Column(Integer, primary_key=True, index=True)
    marca = Column(String(100), nullable=False)
    modelo = Column(String(100), nullable=False)
    anio = Column(Integer, nullable=False)
    matricula = Column(String(20), unique=True, nullable=False)
    color = Column(String(50))
    kilometraje = Column(Integer)
    fecha_ultima_revision = Column(DateTime, default=datetime.utcnow)
    propietario_id = Column(Integer, ForeignKey('usuarios.id'))
    
    # Relaciones
    propietario = relationship("Usuario", back_populates="vehiculos")
    citas = relationship("Cita", back_populates="vehiculo")
    
    def __repr__(self):
        return f"<Vehiculo {self.marca} {self.modelo} - {self.matricula}>" 

File: frontend/src/types/vehiculo.ts
import { Usuario } from './usuario';

export interface Vehiculo {
  id: string;
  marca: string;
  modelo: string;
  anio: number;
  matricula: string;
  kilometraje: number;
  color?: string;
  tipo?: string;
  cliente_id: string;
  created_at?: string;
  updated_at?: string;
}

export interface VehiculoInput {
  marca: string;
  modelo: string;
  anio: number | string;
  matricula: string;
  kilometraje: number | string;
  color?: string;
  tipo?: string;
  cliente_id: string;
}

export interface VehiculoFilter {
  marca?: string;
  modelo?: string;
  anio?: number;
  matricula?: string;
  cliente_id?: string;
} 